frontend/src/components/UserIcon/index.jsx

```diff
@@ -16,6 +18,16 @@ export default function Jazzicon({ size = 10, user, role }) {
     divRef.current.appendChild(result);
   }, [pfp, role, seed, size]);
 
+  useEffect(() => {
+    async function fetchWorkspacePfp() {
+      const pfpUrl = await Workspace.fetchPfp(user?.uid);
+      setWorkspacePfp(pfpUrl);
+    }
+    if (role === "assistant") {
+      fetchWorkspacePfp();
+    }
+  }, [role, user?.uid]);
+
```

Comment:
This will fire hundreds of times on a response chunk and will constantly make a call to the backend for the PFP. On only a small response it called over 80 times. We should be local-caching this image aggressively (maybe even when the workspace is GET on the chat page).

The image request should only be called once and then retrieved from cache then on until some expiration date.


Lastly, UseEffects need to return a function so rewriting this as 
```jsx
useEffect(() => {
    async function fetchWorkspacePfp() {
     if(role !== "assistant") return false;
      const pfpUrl = await Workspace.fetchPfp(user?.uid);
      setWorkspacePfp(pfpUrl);
    }
      fetchWorkspacePfp();
  }, [role, user?.uid]);
  ```
  
  makes more sense here